winch/codegen/src/isa/x64/masm.rs

```diff
@@ -1211,6 +1211,18 @@ impl Masm for MacroAssembler {
 
         Ok(())
     }
+
+    fn wasm_load_atomic(
+        &mut self,
+        src: Self::Address,
+        dst: WritableReg,
+        size: OperandSize,
+        kind: Option<ExtendKind>,
+    ) {
+        // The guarantees of the x86-64 memory model ensure that `SeqCst`
+        // loads are equivalent to normal loads.
```

Comment:
Given this comment and for the purposes of this pull request, I wonder if it's necessary to introduce a new method in the MacroAssembler. I'd prefer if possible, if we could instead extend the current load definition to take in a `load_kind` argument, which categorizes if the load is atomic or not, similar to Chris' comment in https://github.com/bytecodealliance/wasmtime/pull/9970/files#r1909577300, that would also simplify the indirection happening in the `CodeGen` module:

We could simplify to:

* Extending `emit_wasm_load` to take in a load kind
* Performing the right dispatch directly from that method to the MacroAssembler, by passing the in-scope load kind.

Reply:
sure i'll do that